service/matching/task_queue_manager.go

```diff
@@ -371,12 +371,18 @@ func (c *taskQueueManagerImpl) SetInitializedError(err error) {
 	}
 }
 
-func (c *taskQueueManagerImpl) SetUserDataInitialFetch(err error) {
-	c.userDataInitialFetch.Set(struct{}{}, err)
-	if err != nil {
-		// We can't recover from here without starting over, so unload the whole task queue
-		c.lostOwnership.Store(true) // not really lost ownership but we want to skip the last write
-		c.unloadFromEngine()
+// Sets user data enabled/disabled and marks the future ready (if it's not ready yet).
+func (c *taskQueueManagerImpl) SetUserDataInitialFetch(disabledError error, futureError error) {
+	// Always set disabled state even if we're not setting the future.
+	c.db.disableUserData(disabledError)
```

Comment:
This is another misleading statement because it's not necessarily actually disabling user data. What if `disabledError` is nil?

Reply:
Well yes, "disabledError == nil" means "not disabled". Again, it doesn't seem confusing to me but I suppose I did write it :)  I'll try the enum thing